connectors/protocol/connectors.py

```diff
@@ -750,6 +752,19 @@ def validate_configuration_fields(
         missing_fields = list(set(default_config.keys()) - set(current_config.keys()))
 
         if len(missing_fields) > 0:
+            doc = {
+                "last_seen": iso_utc(),
+                "status": "error",
+                "error": f'Configuration fields are missing: {", ".join(missing_fields)}.',
+            }
+            await self.index.update(
+                doc_id=self.id,
+                doc=doc,
+                if_seq_no=self._seq_no,
+                if_primary_term=self._primary_term,
+            )
+            await self.reload()
```

Comment:
I think it should not be done this way - current validation errors make it to the `error` field if they are specified in connector-specific validation method.

Is it possible to rearrange the sync_job_runner code in such a way, that this error in line 768 is caught by same process that marks job as error?

Reply:
@artem-shelkovnikov I've moved the field validation to the `sync_job_runner` code, but left the field property validations in the connector's `prepare` method. It doesn't raise an error and just populates missing field properties so I don't think it's necessary to move. WDYT?

Reply:
To me it sounds good, if it would raise an error and needed to report it, then it would need to be moved to sync_job_runner as well :)